src/applications/Applications.ExtractionLauncher/ExtractionMessageSender.cs

```diff
@@ -0,0 +1,167 @@
+        private readonly bool _isIdentifiableExtraction;
+        private readonly bool _isNoFiltersExtraction;
+        private readonly bool _nonInteractive;
+
+
+        public ExtractionMessageSender(
+            ExtractionLauncherOptions options,
+            ExtractionLauncherCliOptions cliOptions,
+            IProducerModel extractionRequestProducer,
+            IProducerModel extractionRequestInfoProducer,
+            string extractionDir,
+            DateTimeProvider dateTimeProvider,
+            IConsoleInput consoleInput
+        )
+        {
+            _extractionRequestProducer = extractionRequestProducer;
+            _extractionRequestInfoProducer = extractionRequestInfoProducer;
+
+            _extractionDir = (!string.IsNullOrWhiteSpace(extractionDir)) ? extractionDir : throw new ArgumentException(nameof(extractionDir));
+            _dateTimeProvider = dateTimeProvider;
+            _consoleInput = consoleInput;
+
+            _maxIdentifiersPerMessage = options.MaxIdentifiersPerMessage;
+            if (_maxIdentifiersPerMessage <= 0)
+                throw new ArgumentOutOfRangeException(nameof(options.MaxIdentifiersPerMessage));
+
+            _projectId = (!string.IsNullOrWhiteSpace(cliOptions.ProjectId)) ? cliOptions.ProjectId : throw new ArgumentException(nameof(cliOptions.ProjectId));
+            _modalities = cliOptions.Modalities?.ToUpper().Split(',', StringSplitOptions.RemoveEmptyEntries);
+            _isIdentifiableExtraction = cliOptions.IsIdentifiableExtraction;
+            _isNoFiltersExtraction = cliOptions.IsNoFiltersExtraction;
+            _nonInteractive = cliOptions.NonInteractive;
+        }
+
+        public void SendMessages(ExtractionKey extractionKey, List<string> idList)
+        {
+            if (idList.Count == 0)
+                throw new ArgumentException("ID list is empty");
+
+            var jobId = Guid.NewGuid();
+            DateTime now = _dateTimeProvider.UtcNow();
+
+            // TODO(rkm 2021-04-01) Change this to a string[] in both messages below
+            string modalitiesString = _modalities == null ? null : string.Join(',', _modalities);
+
+            var erm = new ExtractionRequestMessage
+            {
+                ExtractionJobIdentifier = jobId,
+                ProjectNumber = _projectId,
+                ExtractionDirectory = _extractionDir,
+                JobSubmittedAt = now,
+                IsIdentifiableExtraction = _isIdentifiableExtraction,
+                IsNoFilterExtraction = _isNoFiltersExtraction,
+
+                // TODO(rkm 2021-04-01) Change this to an ExtractionKey type
+                KeyTag = extractionKey.ToString(),
+
+                Modalities = modalitiesString,
+
+                // NOTE(rkm 2021-04-01) Set below
+                ExtractionIdentifiers = null,
+            };
+
+            var ermList = new List<ExtractionRequestMessage>();
+            for (var i = 0; i <= (idList.Count / _maxIdentifiersPerMessage); ++i)
```

Comment:
Might be more maintainable:

```csharp

/// <summary>
/// Break a list of items into chunks of a specific size
/// </summary>
public static IEnumerable<IEnumerable<T>> Chunk<T>(this IEnumerable<T> source, int chunksize)
{
    while (source.Any())
    {
        yield return source.Take(chunksize);
        source = source.Skip(chunksize);
    }
}
```
https://stackoverflow.com/a/6362642

Reply:
Good catch - will change to that